types/c3/index.d.ts

```diff
@@ -343,171 +439,247 @@ export interface Data {
+         * @param d The data series to decide for.
+         * @returns `false` if selection should be disabled for this data.
+         */
+        isselectable?(this: Record<string, any>, d: DataSeries): boolean;
     };
     /**
      * Set a callback for click event on each data point.
-     * This callback will be called when each data point clicked and will receive d and element as the arguments.
-     * - d is the data clicked and element is the element clicked. In this callback, this will be the Chart object.
+     * @param d The data point that was clicked.
+     * @param element The element for the data point that was clicked.
      */
-    onclick?(d: any, element: any): void;
+    onclick?(this: ChartAPI, d: DataPoint, element: SVGElement): void;
     /**
      * Set a callback for mouseover event on each data point.
-     * This callback will be called when mouse cursor moves onto each data point and will receive d as the argument.
-     * - d is the data where mouse cursor moves onto. In this callback, this will be the Chart object.
+     * @param d The data point that was moused over.
+     * @param element The element for that point. Not passed for some chart types, line 'line'.
      */
-    onmouseover?(d: any, element?: any): void;
+    onmouseover?(this: ChartAPI, d: DataPoint, element?: SVGElement): void;
     /**
      * Set a callback for mouseout event on each data point.
-     * This callback will be called when mouse cursor moves out each data point and will receive d as the argument.
-     * - d is the data where mouse cursor moves out. In this callback, this will be the Chart object.
+     * @param d The data point that the mouse left.
+     * @param element The element for that point. Not passed for some chart types, line 'line'.
      */
-    onmouseout?(d: any, element?: any): void;
-
-    onselected?(d: any, element?: any): void;
+    onmouseout?(this: ChartAPI, d: DataPoint, element?: SVGElement): void;
+    /**
+     * Set a callback for data selection.
+     * @param d The data point that was selected.
+     * @param element The element for the data point that was selected.
+     */
+    onselected?(this: ChartAPI, d: DataPoint, element: SVGElement): void;
+    /**
+     * Set a callback for data deselection.
+     * @param d The data point that was unselected.
+     * @param element The element for the data point that was unselected.
+     */
+    onunselected?(this: ChartAPI, d: DataPoint, element: SVGElement): void;
+    /**
+     * For Stanford charts, specify the key of the epochs data, which maps values to their color.
+     * Defaults to `"epochs"`.
+     */
+    epochs?: string;
 
-    onunselected?(d: any, element?: any): void;
+    /**
+     * Convert data IDs with this function before creating chart.
+     * @param id The original ID string.
+     * @returns The converted ID string.
+     */
+    idConverter?(id: string): string;
 }
 
-export interface Axis {
+export interface AxesOptions {
```

Comment:
Typo? `Axis` with an `i`.

Reply:
That was intentional; "axes" is plural of "axis", so `AxesOptions` contains configuration for each axis. It helps clarify the difference between `AxesOptions` and `AxisConfiguration`.